int32 dbot_m
int32 dbot_itr
